src/main/java/org/triplea/ui/TripleA.java

```diff
@@ -0,0 +1,104 @@
+package org.triplea.ui;
+
+import java.net.URL;
+import java.util.Locale;
+import java.util.ResourceBundle;
+
+import javafx.application.Application;
+import javafx.application.Platform;
+import javafx.fxml.FXML;
+import javafx.fxml.FXMLLoader;
+import javafx.scene.Node;
+import javafx.scene.Scene;
+import javafx.scene.control.Label;
+import javafx.scene.control.Tooltip;
+import javafx.scene.image.Image;
+import javafx.scene.layout.StackPane;
+import javafx.scene.layout.VBox;
+import javafx.stage.Stage;
+
+public class TripleA extends Application {
+
+  @FXML
+  private VBox loadingOverlay;
+
+  @FXML
+  private VBox exitOverlay;
+
+  @FXML
+  private Tooltip progressTooltip;
+
+  @FXML
+  private Label progressLabel;
+
+  @FXML
+  private StackPane rootPane;
+
+  private TripleaMainMenu mainMenu;
+
+  @Override
+  public void start(final Stage stage) throws Exception {
+    final FXMLLoader loader = getLoader(getClass().getResource("./fxml/TripleAMain.fxml"));
+    loader.setController(this);
+    final Scene scene = new Scene(loader.load(), 960, 540);
+    scene.getStylesheets().add("org/triplea/ui/css/main.css");
+    final TripleaMainMenu mainMenu = addRootContent(new TripleaMainMenu(this));
+    this.mainMenu = mainMenu;
+    stage.setMinHeight(scene.getHeight());
+    stage.setMinWidth(scene.getWidth());
+    stage.setScene(scene);
+    stage.getIcons().add(new Image(getClass().getResourceAsStream("/games/strategy/engine/framework/ta_icon.png")));
+    stage.setTitle("TripleA");
+    stage.show();
+    stage.setOnCloseRequest(e -> System.exit(0));
+  }
+
+  public static void launch(final String... arg) {
+    Application.launch(arg);
+  }
+
+  /**
+   * Simplified way of getting an {@link FXMLLoader} with the default settings for TripleA.
+   * @param location The FXML File to load
+   * @return An FXMLLoader object
+   */
+  public static FXMLLoader getLoader(final URL location) {
+    final FXMLLoader loader = new FXMLLoader();
+    loader.setLocation(location);
+    loader.setResources(ResourceBundle.getBundle("org.triplea.ui.lang.TripleA", new Locale("en", "US")));
+    return loader;
+  }
+
+  public TripleaMainMenu getMainMenu() {
+    return mainMenu;
+  }
+
+  public void promptExit() {
+    exitOverlay.setVisible(true);
+  }
+
+  <T extends Node> T addRootContent(final T node) {
+    rootPane.getChildren().add(node);
+    return node;
+  }
+
+  @FXML
+  private void hideExitConfirm() {
+    exitOverlay.setVisible(false);
+  }
+
+  @FXML
+  private void exit() {
+    Platform.exit();
+    System.exit(0);
```

Comment:
Agree with this, and if I remember one of your previous comments, @RoiEXLab, it was due to other application threads keeping the app running?  I don't think this issue needs to be solved in this PR, but it should be a high priority to figure it out relatively soon.

It may be that it's simply an artifact of so many concerns being intertwined, and the JFX client is not inheriting some convoluted piece of code that is (possibly indirectly) responsible for shutting down those threads.

One quick check you could do is run in the debugger and see what threads are still running after the app exits (after removing the `System#exit()` call, of course).  That might give us some clues as to what needs to be done to ensure a clean shutdown from any client.

What the heck, why not just do it, @ssoloff? :smile:  It looks like the only non daemon thread running (other than the JVM shutdown thread) is the AWT/Swing EDT.  Should that thread be running in a JFX app?  Does JFX reuse the AWT/Swing EDT?  I see, what appear to be, JFX-specific threads while the app is running.  You might want to investigate what's going on in `GameRunner#main()` to see if something is inadvertently spinning up the AWT/Swing EDT.

I think it's the `ErrorConsole#getConsole()` call at the top of `main()`.  Commenting that line out, the JFX client exits cleanly without `System#exit()`.